src/main/java/io/lenses/connect/smt/header/TimestampConverter.java

```diff
@@ -296,13 +308,13 @@ public Schema typeSchema(boolean isOptional) {
 
           @Override
           public Date toType(Config config, Date orig) {
-            Calendar result = Calendar.getInstance(UTC);
-            result.setTime(orig);
-            result.set(Calendar.HOUR_OF_DAY, 0);
-            result.set(Calendar.MINUTE, 0);
-            result.set(Calendar.SECOND, 0);
-            result.set(Calendar.MILLISECOND, 0);
-            return result.getTime();
+            ZonedDateTime truncated =
```

Comment:
The translator is responsible for only taking the date part of the incoming Date.

So:

*  orig.toInstant().atZone(config.targetTimeZoneId).toLocalDate(): extracts the date based on the configured timezone
*   LocalTime.MIDNIGHT: it's all about the 0 hour, 0 minute, 0 second, etc
*   UTC:  at this point we don't want another timezone translation because on 01-01 2024 0:0:0 Chicago time would be still in 2023

Reply:
I see the point now, thanks! But IMO it's still too much of a hassle and could be more readable. My proposition:
```
    ZonedDateTime truncated = ZonedDateTime.ofInstant(orig.toInstant(), config.targetTimeZoneId).truncatedTo(ChronoUnit.DAYS);
    return Date.from(truncated.toInstant());
```

Reply:
The option you shared yields incorrect results. in this case it will return the data as of 7am 01/01/1970 - see test testSchemalessTimestampToDateOnNonUTC